site/en/integrations/integrate_with_sentencetransformers.md

```diff
@@ -6,222 +6,176 @@ title: Movie Search Using Milvus and SentenceTransformers
-
-transformer = SentenceTransformer('all-MiniLM-L6-v2')
-
-# Extract the book titles
-def csv_load(file):
-    with open(file, newline='') as f:
-        reader = csv.reader(f, delimiter=',')
-        for row in reader:
-            if '' in (row[1], row[7]):
-                continue
-            yield (row[1], row[7])
-
-
-# Extract embedding from text using OpenAI
-def embed_insert(data):
-    embeds = transformer.encode(data[1]) 
-    ins = [
-            data[0],
-            [x for x in embeds]
-    ]
-    collection.insert(ins)
-
-import time
-
-data_batch = [[],[]]
+model = SentenceTransformer("all-MiniLM-L12-v2")
+```
 
-count = 0
+We loop over the rows of the data, embed the plot summary field, and insert entities into the vector database. In general, you should perform this step over batches of data items to maximize CPU or GPU throughput for the embedding model, as we do here.
 
-for title, plot in csv_load('./movies/plots.csv'):
-    if count <= COUNT:
-        data_batch[0].append(title)
-        data_batch[1].append(plot)
-        if len(data_batch[0]) % BATCH_SIZE == 0:
-            embed_insert(data_batch)
-            data_batch = [[],[]]
-        count += 1
-    else:
-        break
+```python
+for batch in tqdm(ds.batch(batch_size=512)):
+    embeddings = model.encode(batch['PlotSummary'])
+    data = [{"title": title, "embedding": embedding} for title, embedding in zip(batch['Title'], embeddings)]
+    res = collection.insert(data=data)
+```
 
-# Embed and insert the remainder
-if len(data_batch[0]) != 0:
-    embed_insert(data_batch)
+To be safe, we flush the data writing queue and check that the expected number of elements are present in the database.
 
-# Call a flush to index any unsealed segments.
+```python
 collection.flush()
+print(collection.num_entities)
 ```
 
 <div class="alert note">
 
-The above operation is relatively time-consuming because embedding takes time. To keep the time consumed at an acceptable level, try setting `COUNT` in [Global parameters](#Global-parameters) to an appropriate value. Take a break and enjoy a cup of coffee!
+The above operation is relatively time-consuming because embedding takes time. This step takes around 2 minutes using the CPU on a 2023 MacBook Pro and will be much faster with dedicated GPUs. Take a break and enjoy a cup of coffee!
 
 </div>
 
-## Performing the search
-
-With all the data inserted into Milvus, we can start performing our searches. In this example, we are going to search for movies based on the plot. Because we are doing a batch search, the search time is shared across the movie searches. 
+## Performing the Search
+With all the data inserted into Milvus, we can start performing our searches. In this example, we are going to search for movies based on the plot. Because we are doing a batch search, the search time is shared across the movie searches. (Can you guess what the intended result was based on the movie search?)
```

Comment:
what does this mean

(Can you guess what the intended result was based on the movie search?)

i didn't get it. maybe can you guess what the movie is based on the description text as query?